Source/Core/VideoCommon/VertexManagerBase.cpp

```diff
@@ -239,5 +241,53 @@ void VertexManager::Flush()
 
 void VertexManager::DoState(PointerWrap& p)
 {
+	p.Do(ZSlope);
 	g_vertex_manager->vDoState(p);
 }
+
+void VertexManager::CalculateZSlope(u32 stride)
+{
+	float vtx[9];
+	float out[12];
+	float viewOffset[2] = { xfmem.viewport.xOrig - bpmem.scissorOffset.x * 2,
+						    xfmem.viewport.yOrig - bpmem.scissorOffset.y * 2};
+
+	// Lookup vertices of the last rendered triangle and software-transform them
+	// This allows us to determine the depth slope, which will be used if zfreeze
+	// is enabled in the following flush.
+	for (unsigned int i = 0; i < 3; ++i)
+	{
+		u8* vtx_ptr = s_pCurBufferPointer - stride * (3 - i);
```

Comment:
After the vertex loader jit merge, this expression isn't always true. The position may have a offset which is stored within https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoCommon/VertexLoaderManager.h#L28